decorator.rb

```diff
@@ -0,0 +1,24 @@
+require './nameable'
+
+class Decorator < Nameable
+  def initialize(nameable)
+    super()
```

Comment:
I suggest you add  `nameable` as an attribute accessor in the `Decorator` class 
    `attr_accessor :nameable`

Reply:
Thank you :smile: I implemented it :+1: